src/agent/src/main.rs

```diff
@@ -530,13 +525,15 @@ async fn launch_guest_component_procs(
         "spawning confidential-data-hub process {}", CDH_PATH
     );
 
+    tokio::fs::write(OCICRYPT_CONFIG_PATH, OCICRYPT_CONFIG.as_bytes()).await?;
```

Comment:
Where does this config come from? Shouldn't it be processed with initdata?

Reply:
This file is a fixed file and is used by CDH. A better way is to place it in the corresponding directory of guest-image by default when building the guest image. I think this work can be done in conjunction with https://github.com/kata-containers/kata-containers/issues/10979. Before that, put it here (as currently done in [upstream code](https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/main.rs#L584)) could make the logic more dry as they are together.

I'm adding a TODO comment here to add context

Reply:
How about we remove the hard-coded contents and the file writing from here, and 

1. either add a default file during VM build (similar to the default agent policy)
2. or make the hard-coded contents the default for CDH if there is no config on disk

and allow overriding it with initdata?

Feel free to just leave this as a TODO, no need to fix that now.

Reply:
I moved it to the guest-components [build stage](https://github.com/kata-containers/kata-containers/pull/11197/files#diff-2854af34ed318cb07e4fb5c0a50fd52142571ae6b72d036e304454e1e39497bcR38).